functions/sync-review-comments-db/models.py

```diff
@@ -0,0 +1,189 @@
+    INCORRECT = 6
+    VALID_REDUNDANT = 7
+
+
+class ReviewRequestMode(enum.Enum):
+    NORMAL = 1
+    EXPERIMENTAL = 2
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class ReviewRequest(Base):
+    __tablename__ = "review_requests"
+    __table_args__ = (UniqueConstraint("diff_id", "mode", "sequence"),)
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    diff_id: Mapped[int] = mapped_column(index=True)
+    revision_id: Mapped[int] = mapped_column(index=True)
+    status: Mapped[DiffStatus]
+    suggestions: Mapped[List["Suggestion"]] = relationship(
+        back_populates="review_request"
+    )
+    bugbug_version: Mapped[str]
+    tool_variant: Mapped[Optional[str]]
+    ignore_reason: Mapped[Optional[IgnoreReasons]]
+    mode: Mapped[ReviewRequestMode]
+    sequence: Mapped[int]
+
+    # pylint:disable=not-callable
+    created_at: Mapped[datetime] = mapped_column(server_default=func.now())
+    updated_at: Mapped[datetime] = mapped_column(
+        server_default=func.now(), server_onupdate=func.now()
+    )
+
+    @property
+    def is_recently_created(self):
+        return (datetime.utcnow() - self.created_at).total_seconds() < 240
+
+    def has_evaluation(self, session: Session) -> bool:
+        if self.status == DiffStatus.IGNORED:
+            # If the diff is ignored, there's no way it has evaluation. We don't
+            # need to query the database.
+            return False
+
+        stmt = (
+            select(Evaluation.id)
+            .join(Suggestion)
+            .where(Suggestion.review_request_id == self.id)
+            .limit(1)
+        )
+        return session.scalar(stmt) is not None
+
+    def is_replaced(self, session: Session) -> bool:
+        if self.status == DiffStatus.REPLACED:
+            return True
+
+        stmt = select(ReviewRequest.status).where(ReviewRequest.id == self.id)
+
+        return session.scalar(stmt) == DiffStatus.REPLACED
+
+    def suggestions_with_their_evaluation(
+        self, session: Session
+    ) -> ScalarResult["Suggestion"]:
+        stmt = (
+            select(Suggestion)
+            .options(selectinload(Suggestion.evaluation))
+            .where(Suggestion.review_request_id == self.id)
+        )
+        return session.scalars(stmt)
+
+
+class Suggestion(Base):
+    __tablename__ = "suggestions"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    review_request_id: Mapped[int] = mapped_column(ForeignKey(ReviewRequest.id))
+    review_request: Mapped[ReviewRequest] = relationship(back_populates="suggestions")
+    file_path: Mapped[str]
+    line_start: Mapped[int]
+    line_end: Mapped[int]
+    has_added_lines: Mapped[bool]
+    content: Mapped[str]
+    evaluation: Mapped[Optional["Evaluation"]] = relationship(
+        back_populates="suggestion"
+    )
+
+    # TODO: Fill the columns for old suggestions, then change the column
+    # to NOT NULL.
+    llm_name: Mapped[Optional[str]]
+    llm_temperature: Mapped[Optional[float]]
```

Comment:
There are also other parameters, like top_p, and there might be others in the future. How do we handle that?

Reply:
We can add more columns. Alternatively, we can save the metadata as JSON, but it will be less convenient to query it later.

Reply:
The issue I see is in case we add a new parameter, we should make sure to include it here. Can you think of a way to assert that all options are captured?

Reply:
We have the BugBug version and the tool variant, which is a string that describes a group of parameters:

https://github.com/mozilla/bugbug/blob/cf95bc1ae77c9855a8885e93161b572dd79f7678/functions/sync-review-comments-db/models.py#L69-L70

We currently have the following values in the tool variant column: 
- RAG and CONTEXT
- CONTEXT
- RAG

Reply:
I filed an issue to follow up on this: https://github.com/mozilla/bugbug/issues/4734